### DISCUSSION TITLE
A way to change locations of subcaptions?

### DISCUSSION THREAD

**Discussion by victorfeagins** [quarto-dev/quarto-cli#5125] [Q&A] [answered]
I saw there is the `tbl-cap-location` in the yaml to change to location of the main caption. Is there a similar option for subcaptions? 

https://quarto.org/docs/authoring/tables.html#caption-location

**Comment by mcanouil** [comment]
Could you give a bit more context about what you want to achieve?

**Comment by victorfeagins** [comment]
```
#| label: tbl-PostNaive
#| tbl-cap: "Sample Average Treatment Effect: Gender Attitude Post Test"
#| tbl-subcap: "Not Accounting for blocks or clustering"
post.naive.lm <- lm(gender_attitude_post ~ treatment* gender_attitude_base, data = dat)

post.SATE.naive<- avg_comparisons(post.naive.lm,
                variables = list(treatment = c("BAU", "Intervention")))
post.SATE.naive
```

<img width="566" alt="image" src="https://user-images.githubusercontent.com/67335904/230679511-f3fc250b-f1d5-4eda-b64b-d7b5126e46f4.png">

For example that (a) caption if I wanted to move to the bottom of the table instead of the top or even the margin.

**Comment by mcanouil** [comment, marked as answer]
For HTML format, it's pretty easy to achieve the expected results or derivative results.

````qmd
---
title: "Quarto Demo"
format: html
tbl-cap-location: bottom
include-in-header: 
  - text: |
      <style>
      table, .table {
        caption-side: top;
      }
      </style>
---

```{r}
#| label: tbl-PostNaive
#| tbl-cap: "Sample Average Treatment Effect: Gender Attitude Post Test"
#| tbl-subcap: "Not Accounting for blocks or clustering"
knitr::kable(head(iris))
```
````

<img width="750" alt="image" src="https://user-images.githubusercontent.com/8896044/230923567-70df2189-116d-4263-8315-64accec621b8.png">

**Reply by victorfeagins** [reply]
caption-side: bottom to make it show below the table. Thanks!

**Reply by mcanouil** [reply]
Indeed, see <https://developer.mozilla.org/en-US/docs/Web/CSS/caption-side> for other values.

**Comment by maia-sh** [comment]
Thanks for this question @victorfeagins and tips @mcanouil! Unfortunately, I'm having trouble getting this to work for a subcaption for a [subtables](https://quarto.org/docs/authoring/tables.html#subtables) with a [custom cross-reference from Quarto 1.4](https://quarto.org/docs/authoring/cross-references-custom.html). The main table caption is successfully controlled with `supptbl-cap-location`, but `include-in-header` doesn't seem to move the subtable captions. I am far from an HTML expert but I inspected in the rendered HTML and it does seem to be `table.table`, so I'm not sure why it's not being picked up and how to fix it. Thank you for your help!

````qmd
---
title: "My Paper"
format: html
crossref:
  custom:
    - kind: float
      key: supptbl
      reference-prefix:  Supplemental Table
supptbl-cap-location: top
include-in-header: 
  - text: |
      <style>
      table, .table {
        caption-side: top;
      }
      </style>
---

See @supptbl-example.

# Supplement

::: {#supptbl-example layout-nrow=2}
| Col1 | Col2 | Col3 |
|------|------|------|
| A    | B    | C    |
| E    | F    | G    |
| A    | G    | G    |

: First Table

| Col1 | Col2 | Col3 |
|------|------|------|
| A    | B    | C    |
| E    | F    | G    |
| A    | G    | G    |

: Second Table

Main Caption
:::

````

**Reply by mcanouil** [reply]
I don't understand, what's your issue exactly?
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/9a0b58a1-2dfb-461b-a722-7e8887a1cf4e">

**Reply by maia-sh** [reply]
oops! I see my example was not quite right. I swapped to `layout-nrow=2` instead of `ncol`. If this still works for you I can try yet again on my end

**Reply by mcanouil** [reply]
I don't expect this to have been broken in the last week.
